include/fiction/algorithms/physical_design/post_layout_optimization.hpp

```diff
@@ -635,430 +656,182 @@ template <typename Lyt>
     for (uint64_t y = 0; y <= lyt.y(); ++y)
     {
-        bool found_row = true;
-        for (uint64_t x = 0; x <= lyt.x(); ++x)
-        {
-            const bool is_vertical_wire =
-                (lyt.is_wire_tile({x, y}) && lyt.fanin_size(lyt.get_node({x, y})) == 1 &&
-                 lyt.fanout_size(lyt.get_node({x, y})) == 1 && lyt.has_northern_incoming_signal({x, y}) &&
-                 lyt.has_southern_outgoing_signal({x, y}));
-
-            if (!(is_vertical_wire || lyt.is_empty_tile({x, y})))
-            {
-                found_row = false;
-            }
-        }
-        if (found_row)
+        if (!(lyt.is_empty_tile({lyt.x(), y}) ||
+              (lyt.is_po_tile({lyt.x(), y, 0}) && lyt.is_empty_tile({lyt.x() - 1, y + 1, 0}) && (y < lyt.y()))))
         {
-            rows_to_delete.push_back(y);
+            optimizable = false;
         }
     }
 
-    std::vector<uint64_t> columns_to_delete{};
-    columns_to_delete.reserve(lyt.x());
-    for (uint64_t x = 0; x <= lyt.x(); ++x)
+    if (optimizable)
     {
-        bool found_column = true;
-        for (uint64_t y = 0; y <= lyt.y(); ++y)
+        for (uint64_t y = 0; y < lyt.y(); ++y)
         {
-            // Check if the column can be deleted based on certain conditions
-            const bool is_horizontal_wire =
-                (lyt.is_wire_tile({x, y}) && lyt.fanin_size(lyt.get_node({x, y})) == 1 &&
-                 lyt.fanout_size(lyt.get_node({x, y})) == 1 && lyt.has_western_incoming_signal({x, y}) &&
-                 lyt.has_eastern_outgoing_signal({x, y}));
-
-            if (!(is_horizontal_wire || lyt.is_empty_tile({x, y})))
+            if (lyt.is_po_tile({lyt.x(), y, 0}))
             {
-                found_column = false;
+                std::vector<mockturtle::signal<Lyt>> signals{};
+                signals.reserve(lyt.fanin_size(lyt.get_node({lyt.x(), y})));
+                lyt.foreach_fanin(lyt.get_node({lyt.x(), y}),
+                                  [&signals](const auto& fanin) { signals.push_back(fanin); });
+                lyt.move_node(lyt.get_node({lyt.x(), y}), {lyt.x() - 1, y + 1, 0}, signals);
             }
         }
-        if (found_column)
-        {
-            columns_to_delete.push_back(x);
-        }
     }
-
-    delete_rows_and_columns(lyt, rows_to_delete, columns_to_delete);
 }
-
 /**
- * This struct is used to hold information about an update to the layout, where a PO tile
- * is moved from its old coordinate to a new coordinate and connected to its new child node.
+ * Custom comparison function for sorting tiles based on the sum of their coordinates that breaks ties based on the
+ * x-coordinate.
  *
  * @tparam Lyt Cartesian gate-level layout type.
+ * @param a First tile to compare.
+ * @param b Second tile to compare.
+ * @return `true` iff `a < b` based on the aforementioned rule.
  */
 template <typename Lyt>
-struct output_update
+bool compare_gates(const tile<Lyt>& a, const tile<Lyt>& b)
```

Comment:
```suggestion
bool compare_gate_tiles(const tile<Lyt>& a, const tile<Lyt>& b)
```
This naming would better reflect the choice of parameter types accepted by this function.